### ISSUE TITLE
[Feature Request] Give us more control of the reactor through wiring

### ISSUE THREAD

**Issue by Tkachov** [Regalis11/Barotrauma#2232]
// I haven't tried to discuss it anywhere, but I've looked through issues there and haven't seen anything similar.

Last time I checked, reactor's wiring only allowed to shut it down. I had an idea to try to build a [PID controller](https://en.wikipedia.org/wiki/PID_controller) for it, but for that I'd need a way to control fission rate and turbine output through wiring, and there's no such way =( I could've tried [PWM](https://en.wikipedia.org/wiki/Pulse-width_modulation), but there's no way to turn on the reactor, only to shut it down.

----

Another idea, which is more of "just a fun thing to have": somehow allowing moving rods in (and out?) the reactor remotely. The idea is to implement AZ-5 button, which inserts the cooling rods. Again, this is just for fun, so probably isn't worth implementing a way of moving items between containers.

However, if there was a way to control whether reactor slots are "active" (through wiring, of course), such AZ-5 button would just activate slots where player preliminarily put coolant rods. If they put more fuel there, such button would cause meltdown even faster. Plus, controlling "activeness" of slots would allow to remotely control how many rods are burning right now, which is cool too! For example, one could put 4 rods in, but set only one of them to be active. Then, when one rod is burned out, they would just click a button somewhere on the ship, so another slot would activate. I guess through this activation thing one could also implement PWM.

----

TL;DR:
- control reactor's fission rate and turbine output through wiring (or at least turning on);
- a way of moving items between containers (sounds strange and unnecessary, yeah);
- control reactor's slots "activeness" through wiring, so rod isn't burning unless slot is "turned on".

**Comment by Regalis11** [comment]
Setting the charge rate and turbine output through wiring implemented in https://github.com/Regalis11/Barotrauma-development/commit/0d18ef99

**Comment by Regalis11** [comment]
Issue moved to [Regalis11/Barotrauma-development #1738](https://github.com/Regalis11/Barotrauma-development/issues/1738) via [**ZenHub**](https://www.zenhub.com/)

**Comment by Tkachov** [comment]
Thanks for implementing that! And please excuse me for not giving feedback earlier.

I've seen the feature in the current Steam version, but unfortunately haven't got the time to try to make a PID controller I wanted. The only doubt I now have is that I also need grid load and reactor output and I'm not sure where to get them. Anyways, thanks again, and I will post here if I ever manage to make some automatic controller for reactor.
